### PR TITLE
Gql Errors

### PR DESCRIPTION
**Pull request by RichardIrons-neo4j**
Adds GQL compliant errors as per the ADR.

### CODE REVIEW

**Review by AndyHeap-NeoTech — APPROVED** [review]
No obvious changes that I would suggest. Approved.

**Comment by StephenCathcart** [comment]
[[dotnet] Implement GQL Errors as per ADR](https://trello.com/c/dW6FOGTR/471-dotnet-implement-gql-errors-as-per-adr)

File: Neo4j.Driver/Neo4j.Driver.Tests/GqlCompliance/GqlErrorTests.cs [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+        // Assert defaults added by FillGqlDefaults
+        result.GqlDiagnosticRecord.Should().ContainKey("OPERATION");
+        result.GqlDiagnosticRecord["OPERATION"].Should().Be("");
+        result.GqlDiagnosticRecord.Should().ContainKey("OPERATION_CODE");
+        result.GqlDiagnosticRecord["OPERATION_CODE"].Should().Be("0");
+        result.GqlDiagnosticRecord.Should().ContainKey("CURRENT_SCHEMA");
+        result.GqlDiagnosticRecord["CURRENT_SCHEMA"].Should().Be("/");
+
+        result.GqlRawClassification.Should().Be("");
```
Comment (StephenCathcart) [inline Neo4j.Driver/Neo4j.Driver.Tests/GqlCompliance/GqlErrorTests.cs]: It looks like this is causing the test to fail - the actual value is `null` not an empty string.

**Review by StephenCathcart — APPROVED** [review]
Apart from failing test - looks good

**Pull request merged**